content/blogposts/2022/cloud-optimization-part-1-visibility.md

```diff
@@ -0,0 +1,102 @@
+
+Whether you are starting your journey to the cloud, or just looking to better optimize your current cloud costs, it helps to break the problem down into two different buckets.
+
+1. **Infrastructure configuration, optimization, and automation** – Create awareness and visibility into cloud infrastructure usage using infrastructure as code (covered here in part 1).  Then review how your infrastructure can be automated and optimized to find cost savings (covered in part 2 – coming soon).
+2. **Application refactoring** – Review application architecture and resource utilization.  Often applications are monolithic and not designed to be cloud native.  Hyperscalers provide compute, memory, and storage, as well as new standardized services, technologies, and approaches that were not available in the traditional data center environments. This requires insights into the  portions of your code that can leverage the newly available technologies and the ability to rapidly update patterns in code across the organization. At the very least, organizations need to identify and optimize their workloads to run more efficiently in the new hyperscale environment (covered in parts 3 and 4 – coming soon).
+
+### Infrastructure as code 
+Infrastructure as code (IaC) is the process of managing and provisioning cloud infrastructure through machine-readable definition files like Terraform ([open source repositories with examples](https://sourcegraph.com/search?q=context:global+lang:Terraform+select:repo&patternType=literal), [example files](https://sourcegraph.com/search?q=context:global+lang:Terraform+&patternType=literal)) and AWS CloudFormation ([open source repositories with examples](https://sourcegraph.com/search?q=context:global+select:repo+AWSTemplateFormatVersion&patternType=literal), [example files](https://sourcegraph.com/search?q=context:global+AWSTemplateFormatVersion&patternType=literal)), rather than physical hardware configuration or interactive configuration tools [1]. These definition files make it possible to have visibility of all your provisioned infrastructure, especially if you have a [universal code search solution](http://sourcegraph.com) inside your organization.
```

Comment:
@bretthayes For the footnote, is there a way to hyperlink to the footer?

Reply:
Added it in this commit ([840c151](https://github.com/sourcegraph/about/pull/5617/commits/840c15121f476dd49df321ae1d7547e47585f800)) for you! 😄